fem/nonlininteg.hpp

```diff
@@ -79,13 +79,28 @@ class NonlinearFormIntegrator
                                       ElementTransformation &Tr,
                                       const Vector &elfun, Vector &elvect);
 
+   /// Perform the local action (mixed) of the NonlinearFormIntegrator
+   virtual void AssembleElementVector2(const FiniteElement &trial_fe,
```

Comment:
What is the need for this right now and how is it related to this PR? I see you added the abstract method but is there an implementation that is currently been used? If so I suggest we skip this and add it when needed.

Reply:
As I reacted on @kmittal2 above, there is no way how to integrate nonlinear form in the mixed manner currently. It is true some people pass the finite element space through other channels and use `AssembleElementVector()`, but that is nasty 😩 . If we add that this following the interface of `BilinearFormIntegrator` (i.e., `AssembleElementMatrix2`), people may start to use it :wink:. I think this is a relatively good place for this even when not fully related to integration on faces really.